dashboard/lib/projects_list.rb

```diff
@@ -13,13 +24,44 @@ def fetch_section_projects(section)
             # The channel id stored in the project's value field may not be reliable
             # when apps are remixed, so recompute the channel id.
             channel_id = storage_encrypt_channel_id(student_storage_id[:id], project[:id])
-            project_row_data = get_project_row_data(student, project, channel_id)
-            projects_list_data << project_row_data if project_row_data
+            project_data = get_project_row_data(student, project, channel_id)
+            projects_list_data << project_data if project_data
           end
         end
       end
     end
 
+    # Retrieve a hash of lists of published projects from the database, e.g.
+    #   {
+    #     applab: [{...}, {...}, {...}]
+    #   }
+    # when a single project type is requested, or the following when all types
+    # are requested:
+    #   {
+    #     applab: [{...}, {...}, {...}]
+    #     gamelab: [{...}, {...}, {...}]
+    #     playlab: [{...}, {...}, {...}]
+    #     artist: [{...}, {...}, {...}]
+    #   }
+    # @param project_type [String] Type of project to retrieve listed in
+    # PUBLISHED_PROJECT_TYPES, or 'all' to retrieve a list of each project type.
+    # @param limit [Integer] Maximum number of projects to retrieve of each type.
+    #   Must be between 1 and MAX_LIMIT, inclusive.
+    # @param offset [Integer] Number of projects to skip. Default: 0.
+    #   Must not specified when requesting all project types.
+    # @return [Hash<Array<Hash>>] A hash of lists of published projects.
+    def fetch_published_projects(project_type, limit, offset)
```

Comment:
Does it make sense to combine `fetch_published_projects` and `fetch_published_projects_all_types` into one method that accepts a `project_types` array whose default value is `PUBLISHED_PROJECT_TYPES`?